ui/src/Drawer.tsx

```diff
@@ -9,23 +10,67 @@ import SearchResults from "./SearchResults";
 import { ScholarReaderContext } from "./state";
 
 const PDF_VIEWER_DRAWER_OPEN_CLASS = "drawer-open";
-
+const BLACK_LISTED_CLASS_NAME = "MuiTooltip-tooltip";
 export class Drawer extends React.PureComponent {
+  static contextType = ScholarReaderContext;
+  context!: React.ContextType<typeof ScholarReaderContext>;
+
   componentWillUnmount() {
     const { pdfViewer } = this.context;
-    if (pdfViewer !== undefined && pdfViewer !== null) {
+    if (pdfViewer != null) {
       pdfViewer.container.classList.remove(PDF_VIEWER_DRAWER_OPEN_CLASS);
     }
   }
 
+  /**
+   * XXX(zkirby): This is an unfortunate byproduct of the clickaway listener
+   * listening to *all* clicks outside of the drawer. Any developer wishing to 
+   * add a button that opens the drawer will need to add the class name of that element
+   * to the black listed class names list. 
+   */
+  clickAwayClose = (e: React.MouseEvent<Document, MouseEvent>) => {
+    let elementTarget = e.target as (Element | null);
+    while (elementTarget != null) {
+      if (elementTarget.classList.contains(BLACK_LISTED_CLASS_NAME)) {
+        return;
+      }
+      elementTarget = elementTarget.parentElement;
+    }
+
+    this.closeDrawer();
+  }
+
+  closeDrawer = () => {
+    switch(this.drawerState()) {
+      case 'show-symbols': {
+        this.context.setSelectedSymbol(null);
+        break;
+      }
+      case 'show-citations': {
+        this.context.setSelectedCitation(null);
+        break;
+      }
+    }
```

Comment:
Can we combine this with the `closeDrawerOnEscape` functionality in the ScholarReader?

Reply:
I think so! Just to be clear, you mean move the closeDrawerOnEscape into the drawer correct?

Reply:
Yup! Either that, or making some helper function in the ScholarReader or the Drawer that does the setting of the selected citation and symbol to null.